src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -53,20 +53,48 @@ private static bool IsEnabled()
                    s_diagnosticListener.IsEnabled();
         }
 
-        private static Activity? CreateActivity(HttpRequestMessage requestMessage)
+        private static Activity? StartActivity(HttpRequestMessage request)
         {
             Activity? activity = null;
             if (s_activitySource.HasListeners())
             {
-                activity = s_activitySource.CreateActivity(DiagnosticsHandlerLoggingStrings.ActivityName, ActivityKind.Client);
+                Uri? requestUri = request.RequestUri;
+                if (requestUri?.IsAbsoluteUri == false)
+                {
+                    requestUri = null;
+                }
+
+                // The following tags might be important for making sampling decisions and should be provided at span creation time.
+                // https://github.com/open-telemetry/semantic-conventions/blob/5077fd5ccf64e3ad0821866cc80d77bb24098ba2/docs/http/http-spans.md?plain=1#L213-L219
+                KeyValuePair<string, object?>[] tags = requestUri is not null
```

Comment:
Do we have or need a mechanism for Activity to add tags in bulk from a span so that this array allocation could be avoided?

Reply:
Unfortunately no. In the https://github.com/dotnet/runtime/pull/104251#discussion_r1663610456 thread we are discussing avoiding this allocation by dismissing the standard's recommendation, however note that the Activity tag storage is already allocation heavy, see #83345.